jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceWithEntitiesTest.java

```diff
@@ -0,0 +1,239 @@
+
+        // make sure getting mapped variables does NOT throw LazyInitializationException
+        try {
+            varEntity.getMappedVariables();
+        } catch(LazyInitializationException e ) {
+            fail("Unable to retrieve mapped variables : " + e.getMessage());
+        }
+
+    }
+
+    @After
+    public void cleanup() {
+        cleanupSingletonSessionId();
+        if (units != null && !units.isEmpty()) {
+            for (DeploymentUnit unit : units) {
+                try {
+                    deploymentService.undeploy(unit);
+                } catch (Exception e) {
+                    // do nothing in case of some failed tests to avoid next test to fail as well
+                }
+            }
+            units.clear();
+        }
+        close();
+    }
+
+    private static String getPersistenceXml() {
+        return "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n" +
+                "<persistence xmlns=\"http://java.sun.com/xml/ns/persistence\" xmlns:orm=\"http://java.sun.com/xml/ns/persistence/orm\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" version=\"2.0\" xsi:schemaLocation=\"http://java.sun.com/xml/ns/persistence http://java.sun.com/xml/ns/persistence/persistence_2_0.xsd http://java.sun.com/xml/ns/persistence/orm http://java.sun.com/xml/ns/persistence/orm_2_0.xsd\">\n" +
+                "    <persistence-unit name=\"org.jbpm.test:test-module:1.0.0-SNAPSHOT\" transaction-type=\"JTA\">\n" +
+                "        <provider>org.hibernate.ejb.HibernatePersistence</provider>\n" +
+                "        <jta-data-source>java:jdbc/testDS1</jta-data-source>\n" +
+                "        <class>example.CaseDetail</class>\n" +
+                "        <class>org.drools.persistence.jpa.marshaller.MappedVariable</class>\n" +
+                "        <class>org.drools.persistence.jpa.marshaller.VariableEntity</class>\n" +
+                "        <exclude-unlisted-classes>true</exclude-unlisted-classes>\n" +
+                "        <properties>\n" +
+                "            <property name=\"hibernate.dialect\" value=\"org.hibernate.dialect.PostgreSQLDialect\"/>\n" +
+                "            <property name=\"hibernate.max_fetch_depth\" value=\"3\"/>\n" +
+                "            <property name=\"hibernate.hbm2ddl.auto\" value=\"update\"/>\n" +
+                "            <property name=\"hibernate.show_sql\" value=\"false\"/>\n" +
+                "            <property name=\"hibernate.id.new_generator_mappings\" value=\"false\"/>\n" +
+                "            <property name=\"hibernate.transaction.jta.platform\" value=\"org.hibernate.service.jta.platform.internal.JBossAppServerJtaPlatform\"/>\n" +
+                "        </properties>\n" +
+                "    </persistence-unit>\n" +
+                "</persistence>";
+    }
+
+    private static String getCaseDetailEntitySource() {
```

Comment:
Can't this be in a seperate file for better control of the code so it is less error-prone?